src/composables/useCurrentTime.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { createSharedComposable } from '@vueuse/core'
+import { onUnmounted, readonly, ref } from 'vue'
+import type { DeepReadonly, Ref } from 'vue'
+
+/**
+ * Composable to get current time (as Date object)
+ * @param [precision] - precision in milliseconds (defaults to one minute)
+ * @return Date reactive object with current time
+ */
+export function useCurrentTimeComposable(precision: number = 60_000): DeepReadonly<Ref<Date>> {
+	let timeout: NodeJS.Timeout = null
+	let interval: NodeJS.Timeout = null
+
+	const date = ref<Date>(new Date())
+
+	timeout = setTimeout(() => {
+		date.value = new Date()
+		interval = setInterval(() => {
+			date.value = new Date()
+		}, precision)
+	}, precision - date.value % precision)
+
+	// Called for shared composable when all subscribers are unmounted (onScopeDispose)
+	onUnmounted(() => {
+		clearTimeout(timeout)
+		timeout = null
+		clearInterval(interval)
+		interval = null
+	})
+
+	return readonly(date)
+}
+
+/**
+ * Composable to get current time (as Date object)
+ * @return Date reactive object with current time (with 60 seconds precision)
+ */
+function useCurrentTimeMinuteComposable(): DeepReadonly<Ref<Date>> {
+	return useCurrentTimeComposable(60_000)
+}
```

Comment:
Why do we need this function?

Reply:
To not be able to call shared composable with arguments. That way we know all subscribers would be updated once a minute, and not depending on who called first and with what

Reply:
Yes, it is possible in component to call:
`const currentTime = useCurrentTime(1000)`
and update it every second